ubiconfig/_impl/loaders/local.py

```diff
@@ -40,16 +41,21 @@ def load(self, file_name, version=None):
 
                 If version is None, we should get it from its path.
         """
-        if version is None:
-            # get version form path, such as configs/ubi7.1/config.yaml, get
-            # ubi7.1
-            version = os.path.basename(os.path.dirname(file_name))
-
         if not self._isroot:
             file_path = os.path.join(self._path, file_name)
         else:
             file_path = file_name
 
+        if version is None:
+            # get version from path, such as configs/ubi7.1/config.yaml, get
+            # ubi7.1.
+            version = os.path.basename(os.path.dirname(os.path.abspath(file_path)))
+
+        if not re.search("ubi[0-9.]+$", version):
```

Comment:
This does not look correct.
It would match something like ubi7.56.987.98...7, wouldn't it?

Reply:
yeah, it could. 
I was hesitating if I should apply too much restricted pattern. such as ubi[0-9]{1}\.[0-9]{1,2}$, initially I only asked for starting with ubi, but that's too loose, so I changed to this. 

Do you think ubi[0-9]{1}\.[0-9]{1,2}$ would be better?

Reply:
Well, at the very initial, there's no pattern check :(

Reply:
part with dot  must be optional
ubi[0-9]{1}(\.[0-9]{1,2})?

Reply:
yeah.